tests/model/test_model.py

```diff
@@ -513,6 +513,38 @@ def test_append_message_with_valid_log(self, mocker, model):
         (create_msg_box_list.
          assert_called_once_with(model, [0], last_message=expected_last_msg))
 
+    @pytest.mark.parametrize('message', [
+        ({
+            'type': 'private',
```

Comment:
The 'private' and 'stream' are the only changes here, and the rest is duplicated; I would suggest constructing the message in the test and parametrizing based on the type only.